modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductResourceTest.java

```diff
@@ -557,9 +634,96 @@ private void _testPostProductVirtual() throws Exception {
 		Assert.assertNotNull(productVirtualSettingsFileEntry.getSrc());
 	}
 
+	private void _testPostProductWithAccountGroupExternalReferenceCode()
+		throws Exception {
+
+		Product postProduct = productResource.postProduct(
+			_randomProductWithAccountGroup());
+
+		Product randomPatchProduct = _randomProductWithAccountGroup();
+
+		ProductAccountGroup productAccountGroup =
+			randomPatchProduct.getProductAccountGroups()[0];
+
+		long productAccountGroupId = productAccountGroup.getAccountGroupId();
+
+		productAccountGroup.setAccountGroupId(0L);
+
+		randomPatchProduct.setProductAccountGroups(
+			new ProductAccountGroup[] {productAccountGroup});
```

Comment:
probably is better to have the setup of the account groups directly here instead of having the _randomProductWithAccountGroup